src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -204,16 +205,16 @@ public static Result canEntitySpawn(Mob entity, LevelAccessor world, double x, d
         return event.getResult();
     }
 
-    public static boolean canEntitySpawnSpawner(Mob entity, Level world, float x, float y, float z, BaseSpawner spawner)
+    public static boolean canEntitySpawnSpawner(SpawnData spawnData, Mob entity, LevelAccessor world, float x, float y, float z, BaseSpawner spawner)
     {
```

Comment:
Even tho we are not at a RB yet {waiting for Mojang to drop 1.18.2} we're past the point if binary breaks.
So these changes need to be bounced through a deprecated method to keep bin compat.
Also, when adding new arguments to a method, they go at the END to make life a ton less confusing for people calling the method...